### PR TITLE
[fix](partial update) Fix missing rowsets during doing alignment when flushing memtable due to compaction

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes
Currently, the compaction process will delete the old rowset from `_rs_version_map` and not put them to `_stale_rs_version_map` if the version of "to_add" and "to_delete" are exactly same. 
image
image

But we may need some of these old rowsets to do alignment for partial update when flushing memtable. So this PR reset the rowsets needed if we found that the rowsets requested has been deleted when we begin to calculate for merge-on-write table.
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

File: be/src/olap/rowset/beta_rowset_writer.cpp [hunk 1]
```diff
@@ -150,6 +150,14 @@ Status BetaRowsetWriter::_generate_delete_bitmap(int32_t segment_id) {
     {
         std::shared_lock meta_rlock(tablet->get_header_lock());
         specified_rowsets = tablet->get_rowset_by_ids(&_context.mow_context->rowset_ids);
+        _context.mow_context->rowset_ids.clear();
```
Comment (liaoxin01) [inline be/src/olap/rowset/beta_rowset_writer.cpp:153]: line 153 should be removed？

**Comment by bobhan1** [comment]
run buildall

**Review by liaoxin01 — APPROVED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/vertical_segment_writer.cpp [hunk 2]
```diff
@@ -343,6 +343,13 @@ Status VerticalSegmentWriter::_append_block_with_partial_content(RowsInBlock& da
     {
         std::shared_lock rlock(tablet->get_header_lock());
         specified_rowsets = tablet->get_rowset_by_ids(&_mow_context->rowset_ids);
+        if (specified_rowsets.size() != _mow_context->rowset_ids.size()) {
```
Comment (zhannngchen) [inline be/src/olap/rowset/segment_v2/vertical_segment_writer.cpp:346]: The method `_append_block_with_partial_content` of VerticalSegmentWriter should be removed, it's useless for this class, you can submit another pull request to do such cleanup

File: be/src/olap/rowset/beta_rowset_writer.cpp [hunk 3]
```diff
@@ -150,6 +150,13 @@ Status BetaRowsetWriter::_generate_delete_bitmap(int32_t segment_id) {
     {
         std::shared_lock meta_rlock(tablet->get_header_lock());
         specified_rowsets = tablet->get_rowset_by_ids(&_context.mow_context->rowset_ids);
+        if (specified_rowsets.size() != _context.mow_context->rowset_ids.size()) {
+            int64_t cur_max_version = tablet->max_version_unlocked().second;
```
Comment (zhannngchen) [inline be/src/olap/rowset/beta_rowset_writer.cpp:154]: Add a log for such condition

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run clickbench

**Review by zhannngchen — APPROVED** [review]
LGTM
